### ISSUE TITLE
Need an example of Faking IServiceScopeFactory and testing in ExecuteAsync

### ISSUE THREAD

**Issue by aligneddev** [ardalis/CleanArchitecture.WorkerService#3]
In the existing EntryPointServiceExecute test, null is passed into the `new EntryPointService(.....)`.

How do I fake IServiceScopeFactory, so that I can test pieces inside of the using in `ExecuteAsync` of EntryPointService?

```
using (var scope = _serviceScopeFactory.CreateScope())
{
  // test my code here...
```

**Comment by aligneddev** [comment]
This suggests a few options, I like the idea of hiding it behind an abstraction of IServiceLocator: https://stackoverflow.com/a/53509491/54288

thoughts?

**Comment by ardalis** [comment]
Closed via #4
